src/openms/source/CHEMISTRY/EnzymaticDigestion.cpp

```diff
@@ -115,6 +115,65 @@ namespace OpenMS
     return positions;
   }
 
+  Size EnzymaticDigestion::semiSpecificDigestion_(const std::vector<int>& cleavage_positions, std::vector<std::pair<Size, Size>>& output, Size min_length, Size max_length) const
+  {
+    int mc = missed_cleavages_;
+    Size wrong = 0;
+
+    // For every position, add an "extension" of it until next cleavage, in both directions.
+    // This is done by iterating through sequence from both ends at the same time.
+    // Warning: cleavage_positions array has to be sorted.
+    // Warning: this assumes first and last cleavages to be sequence termini.
+
+    const int first_c = cleavage_positions.front(); // First cleavage
+    const int last_c = cleavage_positions.back(); // Last cleavage
+    const int lgth = last_c - first_c;
+
+    // Lambda checking min & max conditions and adding to output
+    auto variant = [&output, &wrong, min_length, max_length](Size x, Size y)
+    {
+      if (min_length <= std::abs((int)y - (int)x) &&
+          max_length >= std::abs((int)y - (int)x))
+      {
+        output.emplace_back(x, y);
+      } else ++wrong;
+    };
+
+    auto fwd = cleavage_positions.cbegin();
+    auto bwd = cleavage_positions.crbegin(); // Reverse iterator!
+    ++fwd; ++bwd;
```

Comment:
dangerous!
check `cleavage_positions.empty()` and return as a first action in the function.